merger/merger.go

```diff
@@ -484,52 +509,34 @@ func shouldKeep(e bf.Expr) bool {
 	return false
 }
 
-// match looks for the matching CallExpr in stmts using X and name
-// i.e. two 'go_library(name = "foo", ...)' are considered matches
-// despite the values of the other attributes.
-// exception: if c is a 'load' statement, the match is done on the first value.
-func match(stmts []bf.Expr, c *bf.CallExpr) (int, *bf.CallExpr) {
-	var m matcher
-	if kind := kind(c); kind == "load" {
-		if len(c.List) == 0 {
-			return -1, nil
-		}
-		m = &loadMatcher{stringValue(c.List[0])}
-	} else {
-		m = &nameMatcher{kind, name(c)}
-	}
+// match looks for the matching CallExpr in stmts. If a rule with a matching
+// "name" attribute is found, and the kind matches, the rule and its index
+// are returned. If a rule with a matching name is found but the kind does not
+// match, the index of the rule and nil are returned. If no match is found,
+// -1 and nil are returned.
+func match(stmts []bf.Expr, x *bf.CallExpr) (int, *bf.CallExpr) {
```

Comment:
The api is not great, you can't do _, rule = match(...) when you don't care about type.
Not needed to fix for this PR (it's a private rarely called function) but consider changing to something that maybe returns index, rule, bool where index and rule are both always set or not together, and the final bool indicates a full or partial match (or something)

Reply:
Done, that seems sensible.

This function will need to change in the future anyway, since we'll need to match on more attributes (e.g., importpath).